src/pymodaq/daq_utils/daq_utils.py

```diff
@@ -442,11 +442,37 @@ def uncapitalize(string, Nfirst=1):
     return string[:Nfirst].lower() + string[Nfirst:]
 
 
-def get_data_dimension(arr, scan_type='scan1D', remove_scan_dimension=False):
-    dimension = len(arr.shape)
-    if dimension == 1:
-        if arr.size == 1:
-            dimension = 0
+def get_data_dimension(data: np.ndarray, scan_type='scan1D', remove_scan_dimension=False):
+    """Return the shape, dimension and size of the input data.
+
+    Notice that the dimension returned does not necessarily correspond to the size of the input ndarray. If the shape of
+    the ndarray is (1,) it is interpreted as a scalar data, thus of dimension 0. This is done because it seems to be the
+    way pytables library manages scalar data (https://www.pytables.org/_modules/tables/earray.html#EArray.append).
+
+    Parameters
+    ----------
+    data : numpy ndarray
+        Typically raw data from an acquisition (a spectrum, an image…).
+    scan_type : str
+        ??
+    remove_scan_dimension : bool
+        ??
+
+    Returns
+    -------
+    tuple of tuple, string, int
+
+    """
+    if not isinstance(data, np.ndarray):
+        raise Exception("Input parameter should be an ndarray.")
+
+    if data.ndim == 0:
+        raise Exception("The dimension of the array should be superior to zero.")
```

Comment:
Same, `ValueError` could be used here instead.

Reply:
Done.

Reply:
Here: cd169f35d12eae3e0ae9ed0f6d162853d90264d7.